Objects/longobject.c

```diff
@@ -167,78 +160,101 @@ _PyLong_Copy(PyLongObject *src)
     return (PyObject *)result;
 }
 
-/* Create a new int object from a C long int */
+static PyObject *
+_PyLong_FromMedium(sdigit x)
+{
+    assert(!IS_SMALL_INT(x));
+    assert(IS_MEDIUM_INT(x));
+    /* We could use a freelist here */
+    PyLongObject *v = PyObject_Malloc(sizeof(PyLongObject));
+    if (v == NULL) {
+        PyErr_NoMemory();
+        return NULL;
+    }
+    Py_ssize_t sign = x < 0 ? -1: 1;
+    digit abs_x = x < 0 ? -x : x;
+    _PyObject_InitVar((PyVarObject*)v, &PyLong_Type, sign);
+    v->ob_digit[0] = abs_x;
+    return (PyObject*)v;
+}
 
-PyObject *
-PyLong_FromLong(long ival)
+static PyObject *
+_PyLong_FromLarge(stwodigits ival)
```

Comment:
OK, I assumed that `sizeof(stwodigit) >= sizeof(long)` would always be true. I'll revert `PyLong_FromLong`.

OOI, why do we allow `digits` to be configured, rather than being derived from the word size?

Reply:
> OOI, why do we allow `digits` to be configured, rather than being derived from the word size?

Ah, that one's my fault. When I introduced the 30-bit digits, it wasn't clear at the time what platforms this would give a speedup on, so I wanted to leave an option so that platforms could choose one way or the other depending on their own perfomance tests. It did also make it rather easier to test the 15-bit digit behaviour.

The 30-bit digits were also dependent on the existence of `long long`, which we weren't allowed to assume back then.

I've thought several times that we should just dump the option of 15-bit digits and use 30-bit digits everywhere. I'm fairly sure that would be the right choice for all current laptop and desktop machines. I don't know whether the 15-bit digit option has value for embedded scenarios, though.

Reply:
(Also, while we're musing about digit size, 63-bit digits may make more sense than 60-bit digits; the multiple-of-5 constraint comes from two sources - the marshal format, and the power algorithm, but I dare say we could adapt both. Anyway, this is getting off-topic for this PR.)